tests/run/i7788.scala

```diff
@@ -0,0 +1,14 @@
+trait Show[-A] with
+  def show(a:A): String
+
+given Show[String] = x => x
+given Show[Int] = _.toString
+
+given showEither[A,B]: (sA: Show[A]) => Show[B] => Show[Either[A,B]] =
+  _.fold(a => s"Left(${summon[Show[A]].show(a)})", b => s"Right(${summon[Show[B]].show(b)})")
+given [A,B]: (sA: Show[A]) => (sB: Show[B]) => Show[(A,B)] = (a,b) => s"(${sA.show(a)}), ${sB.show(b)})"
+
+@main def ShowDemo =
```

Comment:
```suggestion
@main def Test =
```
failing CI because can't find class Test